packages/react-scripts/config/env.js

```diff
@@ -29,7 +29,12 @@ function getClientEnvironment(publicUrl) {
       // For example, <img src={process.env.PUBLIC_URL + '/img/logo.png'} />.
       // This should only be used as an escape hatch. Normally you would put
       // images into the `src` and `import` them in code to get their paths.
-      'PUBLIC_URL': publicUrl
+      'PUBLIC_URL': publicUrl,
+      // Useful for allowing the hot dev websocket to connect to the host
+      // specified in the env file, instead of window.location, which could
+      // be wrong if the dev server is being proxied.
+      'HOST': process.env.HOST || '',
+      'PORT': process.env.PORT || ''
```

Comment:
I believe it needs to be configured somehow. The .env file was the only place that I knew of that allowed this, that's why I chose to put it there.

I wasn't aware that variables could be shared in another way but if you point me in the direction of an example I could change this.